edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -41,6 +47,8 @@ public AvailabilityMetrics(IMetricsProvider metricsProvider, ISystemTime time)
             this.time = time;
             this.availabilities = new List<Availability>();
             this.edgeAgent = new Lazy<Availability>(() => new Availability("edgeAgent", this.CalculateEdgeAgentDowntime(), this.time));
+
+            this.checkpoint = new PeriodicTask(this.NoteCurrentTime, this.checkpointFrequency, this.checkpointFrequency, this.log, "Checkpoint Availability");
         }
 
         public void ComputeAvailability(ModuleSet desired, ModuleSet current)
```

Comment:
below current.Modules.Values.Where(c => (c is IRuntimeModule)) should extract and reuse.  e.g. current.Modules.Values.Select(c => c as IRuntimeModule).Where(m => m != null)

Reply:
line 72, you should use Constants.EdgeAgentModuleName to replace "edgeAgent"